vms/evm/contract_message.go

```diff
@@ -23,13 +24,22 @@ func NewContractMessage(logger logging.Logger, subnetInfo config.SourceSubnet) *
 }
 
 func (m *contractMessage) UnpackWarpMessage(unsignedMsgBytes []byte) (*vmtypes.WarpMessageInfo, error) {
+	// This function may be called with raw UnsignedMessage bytes or with ABI encoded bytes as emitted by the Warp precompile
+	// The latter case is the steady state behavior, so check that first. The former only occurs on startup.
```

Comment:
Definitely open to feedback on this... When taken from an event log's data, `unsignedMsgBytes` is prefixed by some additional metadata (that I believe comes from ABI encoding the unsigned message bytes). 

I'm sure we could trim this data if it exists in all cases, but that would slightly increase the amount of computation in the normal operation case. Here, the execution time for the normal operation case is unaffected.

Reply:
Yeah, it's more of a stylistic thing than anything - keeping the normal case on the shortest path is good practice.

I updated the error logging so that we only emit an error if both cases fail. I kept the Warn log, but bumped it down to Debug so that we have a way to measure one failling but not the other. I also elected to keep this check inline since it's only used in one place, and the individual cases are already outsourced to utilities.